server/src/main/java/org/candlepin/resource/UserResource.java

```diff
@@ -155,7 +156,7 @@ public UserDTO getUserInfo(@PathParam("username") @Verify(User.class) String use
             // to avoid leaking role details about other users.
             return roles.stream()
                 .map(this.modelTranslator.getStreamMapper(RoleInfo.class, RoleDTO.class))
-                .peek(e -> e.setUsers(users));
+                .peek(e -> e.setUsers(users.stream().collect(Collectors.toList())));
```

Comment:
You can use ArrayList constructor instead of going through stream.
```java
new ArrayList<UserDTO>(users)
```
Second point. I know it was already there, but peek is intended mainly for debugging. Map would be much clearer.
```java
.map(e -> e.users(new ArrayList<>(users)));
```

Reply:
Updated to Map.